File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return answer;
    }
    
    private void dfs(int[] numbers, int target, int len, int now) {
        if(len == numbers.length) {
            if(now == target) answer++;
            return;
        }
        dfs(numbers, target, len + 1, now + numbers[len]);
        dfs(numbers, target, len + 1, now - numbers[len]);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    def dfs(temp,index):
        if index==len(numbers):
            if temp==target:
                return 1
            else: return 0
        return dfs(temp+numbers[index], index+1)+dfs(temp-numbers[index], index+1)
    return dfs(0,0)